### PR TITLE
Use getOnboardingState api for dedicated setup flow

### PR DESCRIPTION
**Pull request by selfcontained**
## Description

Hooks up the new `getOnboardingState()` api to our check for showing the dedicated setup. I had to add a bit of client-side state tracking for when we show/complete the setup flow. I also made a couple other changes.

* Filtering out deleted oidc clients from `hasAnyOrgWithActiveSSO()`
* Removing the caching and ff check in `getOnboardingState()`

## Related Issue(s)

Fixes WEB-245

## How to test

* Try and login to the preview env with your gitpod.io email: https://bmh-dedica741b327561.preview.gitpod-dev.com/workspaces
* If you're an admin/owner of the org, you can also force the setup flow manually by adding `?dedicated-setup=force` to the url and reload.

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - bmh-dedica741b327561</li>
	<li><b>🔗 URL</b> - <a href="https://bmh-dedica741b327561.preview.gitpod-dev.com/workspaces" target="_blank">bmh-dedica741b327561.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [x] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

File: components/dashboard/src/dedicated-setup/DedicatedSetup.tsx [hunk 1]
```diff
@@ -78,8 +80,9 @@ const DedicatedSetupSteps: FC<DedicatedSetupStepsProps> = ({ org, config }) => {
     }, [dropConfetti]);
 
     const handleEndSetup = useCallback(() => {
+        markCompleted();
```
Comment (AlexTugarev) [inline components/dashboard/src/dedicated-setup/DedicatedSetup.tsx:83]: Q: I recall the question about the session which is replaced at the end of a completed OIDC flow. Do we still need to reset some caches as the ID of the user is about to change on completion of the onboarding?

Comment (selfcontained) [reply components/dashboard/src/dedicated-setup/DedicatedSetup.tsx]: Yah, I've added that [here](https://github.com/gitpod-io/gitpod/pull/17488/files#diff-658b622be42a8e93d942c1349dfc9ff9a996dff6f72a940ac81a126afd8e0b21R84).

File: components/dashboard/src/dedicated-setup/use-check-dedicated-setup.ts [hunk 2]
```diff
@@ -4,20 +4,61 @@
...
+        noPersistence(["onboarding-state"]),
+        async () => {
+            return getGitpodService().server.getOnboardingState();
+        },
+        {
+            // Cache for a bit so we can avoid having the value change before we're ready
+            staleTime: 1000 * 60 * 60 * 1, // 1h
+            cacheTime: 1000 * 60 * 60 * 1, // 1h
+            // Only query if feature flag is enabled
+            enabled: enableDedicatedOnboardingFlow,
```
Comment (AlexTugarev) [inline components/dashboard/src/dedicated-setup/use-check-dedicated-setup.ts:61]: This should be sufficient to turn it off on gitpod.io effectively. 👍🏻

File: components/gitpod-db/src/typeorm/team-db-impl.ts [hunk 3]
```diff
@@ -384,6 +384,7 @@ export class TeamDBImpl implements TeamDB {
         const result = await repo.query(
             `select org.id from d_b_team as org inner join d_b_oidc_client_config as oidc on org.id = oidc.organizationId
                 where oidc.active = 1
+                and oidc.deleted = 0
```
Comment (AlexTugarev) [inline components/gitpod-db/src/typeorm/team-db-impl.ts:387]: 👍🏻

File: components/dashboard/src/dedicated-setup/OrgNamingStep.tsx [hunk 4]
```diff
@@ -28,7 +28,16 @@ export const OrgNamingStep: FC<Props> = ({ onComplete }) => {
         if (org.data) {
             updateOrg.mutate({ name: orgName }, { onSuccess: onComplete });
         } else {
-            createOrg.mutate({ name: orgName }, { onSuccess: onComplete });
+            createOrg.mutate(
+                { name: orgName },
+                {
+                    onSuccess: (newOrg) => {
+                        // Need to manually set active-org here so it's returned via subsequent useCurrentOrg() calls
+                        localStorage.setItem("active-org", newOrg.id);
```
Comment (svenefftinge) [inline components/dashboard/src/dedicated-setup/OrgNamingStep.tsx:36]: In other places we do a redirect, in order to achieve this, which seems to be rather destructive method. Would it make sense to expose a `useChangeCurrentOrg` hook from the org-query module?

Comment (svenefftinge) [reply components/dashboard/src/dedicated-setup/OrgNamingStep.tsx:36]: what triggers the loading of the changed value?

**Review by svenefftinge — APPROVED** [review]
LGTM

**Review by AlexTugarev — COMMENTED** [review]
Looks great! Testing once more...

**Comment by AlexTugarev** [comment]
There were two issues:
* (less critical) there is a bit of flickering on load of the onboarding view. :ok_hand:
* after verifying the SSO config, I do see the confetti, but there is no forwarding to the next page (Org settings) :disappointed:
   * I navigated to /workspaces eventually, and could see that the setup is completed, i.e. the Org + SSO config are present and working. Did log out and log in again :heavy_check_mark:

**Comment by AlexTugarev** [comment]
We applied a fix for the missing Complete step together with @svenefftinge, and tested in a parallel preview env.
See internal [thread](https://gitpod.slack.com/archives/C02EN94AEPL/p1683191364669679?thread_ts=1683159600.109259&cid=C02EN94AEPL).

**Comment by AlexTugarev** [comment]
/unhold

🤞🏻  for the build to be green.

**Pull request merged**